src/foundry/foundry.js/applications/formApplications/documentSheets/combatantConfig.d.ts

```diff
@@ -0,0 +1,57 @@
+import { ConfiguredDocumentClass } from '../../../../../types/helperTypes';
+
+declare global {
+  /**
+   * The Application responsible for configuring a single Combatant document within a parent Combat.
+   *
+   * @typeParam Options - the type of the options object
+   * @typeParam Data    - The data structure used to render the handlebars template.
+   */
+  class CombatantConfig<
+    Options extends DocumentSheet.Options = CombatantConfig.Options,
+    Data extends CombatantConfig.Data<Options> = CombatantConfig.Data<Options>
+  > extends DocumentSheet<Options, Data, InstanceType<ConfiguredDocumentClass<typeof Combatant>>> {
+    /**
+     * @defaultValue
+     * ```typescript
+     * foundry.utils.mergeObject(super.defaultOptions, {
+     *   id: "combatant-config",
+     *   title: game.i18n.localize("COMBAT.CombatantConfig"),
+     *   classes: ["sheet", "combat-sheet"],
+     *   template: "templates/sheets/combatant-config.html",
+     *   width: 420
+     * });
+     * ```
+     */
+    static get defaultOptions(): CombatantConfig.Options;
+
+    /** @override */
+    get title(): string;
+
+    /**
+     * @param event - (unused)
+     * @override
+     */
+    protected _updateObject(
+      event: Event,
+      formData: CombatantConfig.FormData
+    ): Promise<InstanceType<ConfiguredDocumentClass<typeof Combatant>> | undefined>;
+  }
+
+  namespace CombatantConfig {
+    interface Options extends DocumentSheet.Options {
+      id: 'combatant-config';
+      title: string;
+      classes: string[];
+      template: string;
+      width: number;
+    }
+
+    type Data<Options extends DocumentSheet.Options> = DocumentSheet.Data<
```

Comment:
I don't think we actually need this type, do we? It might make sense to include it for brevity in the declaration of `CombatantConfig`, so either way is fine. Just make a conscious decision about it :)

Reply:
My motivation behind at was to be more consistent with the other sheets. If you override multiple sheets and have a `SheetClass.Data` on each and no `CombatantConfig.Data` this might be confusing.

Reply:
Then it’s fine